### ISSUE TITLE
IntelliSense variable inspection on hover too slow

### ISSUE THREAD

**Issue by nonsensation** [OmniSharp/omnisharp-vscode#4953]
## Issue Description ##
While debugging - hover on variable to see its value takes too long.

## Steps to Reproduce ##
1. New project `dotnet new console`
2. Simple local variable `var test = "123";`
3. Set a breakpoint after variable, so its initialized
4. Start debugger (F5)
5. Hover over variable

## Expected Behavior ##
Show me the value in an instant.
First time might be slower, but consecutive times thereafter should show the value as soon as I hover over the variable.

## Actual Behavior ##
It  takes ~3+ seconds to show the value, even for simple ones as ints and strings in an minimal project.
It also takes ~3 Seconds every time when hovered, not just the first time.

---

I haven't had this problem roughly half a year ago, as far as I can remember, and just came back to VSCode, updated it to the latest version, installed .NET 6.0 (also 5.0 to check if it is .NET), disabled all extensions, searched for this issue on google/github and searched for settings, as it may be a artificial delay.
When not debugging, the tooltip shows up as expected and configured in the settings (`"editor.hover.delay": 1000,`), however while debugging it is way too slow.

### OmniSharp log ###
<details>

```
Starting OmniSharp server at 13/12/2021, 09:52:31
    Target: d:\Code

OmniSharp server started.
    Path: c:\Users\3li\.vscode\extensions\ms-dotnettools.csharp-1.23.17\.omnisharp\1.37.18-beta.111\OmniSharp.exe
    PID: 8688

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on Windows 6.2.9200.0 (x64)
[info]: OmniSharp.Services.DotNetCliService
        Checking the 'DOTNET_ROOT' environment variable to find a .NET SDK
[info]: OmniSharp.Services.DotNetCliService
        Using the 'dotnet' on the PATH.
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 17.0.0 - "c:\Users\3li\.vscode\extensions\ms-dotnettools.csharp-1.23.17\.omnisharp\1.37.18-beta.111\.msbuild\Current\Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to 'c:\Users\3li\.vscode\extensions\ms-dotnettools.csharp-1.23.17\.omnisharp\1.37.18-beta.111\.msbuild\Current\Bin\MSBuild.exe'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 17.0.0 - "c:\Users\3li\.vscode\extensions\ms-dotnettools.csharp-1.23.17\.omnisharp\1.37.18-beta.111\.msbuild\Current\Bin"
            CscToolExe = csc.exe
            CscToolPath = c:\Users\3li\.vscode\extensions\ms-dotnettools.csharp-1.23.17\.omnisharp\1.37.18-beta.111\.msbuild\Current\Bin\Roslyn
            MSBuildExtensionsPath = c:\Users\3li\.vscode\extensions\ms-dotnettools.csharp-1.23.17\.omnisharp\1.37.18-beta.111\.msbuild
            MSBuildToolsPath = c:\Users\3li\.vscode\extensions\ms-dotnettools.csharp-1.23.17\.omnisharp\1.37.18-beta.111\.msbuild\Current\Bin
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.Completion.CompletionOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in 'd:\Code'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Did not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in 'd:\Code'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for 'd:\Code\AdventOfCode\2021\AdventOfCode_2021.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for 'd:\Code\Test\Test.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in 'd:\Code'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Did not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location 'd:\Code' on host 2864.
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: d:\Code\AdventOfCode\2021\AdventOfCode_2021.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file 'd:\Code\AdventOfCode\2021\AdventOfCode_2021.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project 'd:\Code\AdventOfCode\2021\AdventOfCode_2021.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: d:\Code\Test\Test.csproj
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file 'd:\Code\Test\Test.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project 'd:\Code\Test\Test.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: AdventOfCode_2021
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: Test
[info]: OmniSharp.Roslyn.CSharp.Services.Diagnostics.CSharpDiagnosticWorkerWithAnalyzers
        Solution initialized -> queue all documents for code analysis. Initial document count: 12.
Received response for /v2/codestructure but could not find request.
Received response for /v2/getcodeactions but could not find request.
```

</details>

## Environment information ##

**VSCode version**: 1.63.0
**C# Extension**: 1.23.17

<details><summary>Dotnet Information</summary>
.NET SDK (reflecting any global.json):
 Version:   6.0.100
 Commit:    9e8b04bbff

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19043
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100\

Host (useful for support):
  Version: 6.0.0
  Commit:  4822e3c3aa

.NET SDKs installed:
  5.0.403 [C:\Program Files\dotnet\sdk]
  6.0.100 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|csharp|ms-dotnettools|1.23.17|;

</details>

**Comment by crystalbrittle** [comment]
I have this same issue debugging javascript. ~3 second delay. Also unaffected by the `"editor.hover.delay"` setting. I would love a way to make this much faster, independent of the `"editor.hover.delay"` setting.

**Comment by RickvdLaan** [comment]
I actually have the same issues on my personal computer at home, haven't found a solution yet.

**Comment by nonsensation** [comment]
I have searched for a long time how to fix this, as it is so annoying to wait over a second to watch the value of a float.
I have digged into hover-settings even the source of vscode if I could find ANYTHING related, but no success at all.

Then I stumbled upon [this issue](https://github.com/microsoft/vscode/issues/108333) #108333 , where @isidorn mentioned, the Quick Watch Window (as seems to be called) has been rewritten at that time around.
So I changed in VSCode the C# extension (little gear in extension panel) to an Install another version..** (1.21.12 - 2 yrs ago). As usual it says a restart is required. After that restart, it told me again a restart is required, I pressed it again.
Suddenly I have the latest version installed (1.24.0) and the hover is fast and quick again.
I don't know what happened or how it was fixed, but it seems to work for now. Tbh, I am afraid of testing any more as it might become annoingly slow again.
